### PR TITLE
feat: add `create-t3-app` spec

### PR DESCRIPTION
**Pull request by fernandoxu**
Feature add [create-t3-app](https://create.t3.gg/)

### CODE REVIEW

File: src/create-t3-app@latest.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const dirArgument: Fig.Arg = {
+  name: "dir",
+  description:
+    "The name of the application, as well as the name of the directory to create",
+  template: "folders",
+  isOptional: true,
+};
+
+const completionSpec: Fig.Spec = {
+  name: "create-t3-app@latest",
```
Comment (fedeci) [inline src/create-t3-app@latest.ts:10]: Can you remove @latest please (from the filename also)? Eventually if you are used to run this CLI with the trailing @latest you can create another file named `src/create-t3-app@latest.ts` that contains the following:
```ts
import create_t3_app from "./create_t3_app";

const completionSpec: Fig.Spec = {
  ...create_t3_app,
  name: "create_t3_app@latest",
};

export default completionSpec;
```

File: src/create-t3-app@latest.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const dirArgument: Fig.Arg = {
+  name: "dir",
+  description:
+    "The name of the application, as well as the name of the directory to create",
+  template: "folders",
```
Comment (fedeci) [inline src/create-t3-app@latest.ts:5]: Do folders need to exist or they may not exist?

File: src/npx.ts [hunk 3]
```diff
@@ -75,6 +75,10 @@ const suggestions: Fig.Suggestion[] = [
     name: "create-next-app",
     icon: "https://nextjs.org/static/favicon/favicon-16x16.png",
   },
+  {
+    name: "create-t3-app@latest",
+    icon: "https://create.t3.gg/favicon.svg",
+  },
```
Comment (fedeci) [inline src/npx.ts:81]: Here you should duplicate the suggestion adding "create-t3-app" also.

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks! Great first spec!

File: src/create-t3-app@latest.ts (hunk 1 above)
Comment (fernandoxu) [reply src/create-t3-app@latest.ts]: OK! thanks a lot for your check!

File: src/create-t3-app@latest.ts (hunk 2 above)
Comment (SeparateRecords) [reply src/create-t3-app@latest.ts]: IIRC you're correct, folder doesn't need to exist

You can add `suggestCurrentToken: true` to make the behavior correct! (I think that's the name of the prop...)

File: src/create-t3-app.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      name: ["-y", "--default"],
+      description:
+        "Bypass the CLI and use all default options to bootstrap a new t3-app (default: false)",
+    },
+    {
+      name: ["--help", "-h"],
+      description: "Display help for command",
+    },
+  ],
+};
```
Comment (fedeci) [inline src/create-t3-app.ts:21]: This wasn't to remove, we have options for everything! `suggestCurrentToken` was what this argument needed.
```suggestion
const dirArgument: Fig.Arg = {
  name: "dir",
  description:
    "The name of the application, as well as the name of the directory to create",
  template: "folders",
  isOptional: true,
  suggestCurrentToken: true,
};

const completionSpec: Fig.Spec = {
  name: "create-t3-app",
  description: "A CLI for creating web applications with the t3 stack",
  icon: "https://create.t3.gg/favicon.svg",
  options: [
    {
      name: "--noGit",
      description:
        "Explicitly tell the CLI to not initialize a new git repo in the project (default: false)",
    },
    {
      name: ["-y", "--default"],
      description:
        "Bypass the CLI and use all default options to bootstrap a new t3-app (default: false)",
    },
    {
      name: ["--help", "-h"],
      description: "Display help for command",
    },
  ],
  args: dirArgument,
};
```

**Review by fedeci — CHANGES_REQUESTED** [review]

**Review by fedeci — APPROVED** [review]
Thanks again!

**Pull request merged**